flink/src/main/java/org/apache/iceberg/flink/data/FlinkOrcWriter.java

```diff
@@ -91,6 +91,12 @@ private WriteBuilder() {
         case BOOLEAN:
           return GenericOrcWriters.booleans();
         case INTEGER:
+          switch (flinkPrimitive.getTypeRoot()) {
+            case TINYINT:
+              return GenericOrcWriters.bytes();
```

Comment:
This fixed the broken unit tests in flink: https://github.com/apache/iceberg/pull/1255/commits/2ec4ce6030230e90e8becf3b0a79dca88c4024e1#diff-6820fc22b4e5cbfa4a1c029bf5c8c789L255.  

we may need to add similar UT in spark so that we could write the tinyint and smallint to spark correctly,  I will take a look.